erpnext/buying/doctype/request_for_quotation/request_for_quotation.py

```diff
@@ -394,11 +394,11 @@ def get_item_from_material_requests_based_on_supplier(source_name, target_doc =
 
 @frappe.whitelist()
 def get_supplier_tag():
-	if not frappe.cache().hget("Supplier", "Tags"):
-		filters = {"document_type": "Supplier"}
-		tags = list(set(tag.tag for tag in frappe.get_all("Tag Link", filters=filters, fields=["tag"]) if tag))
-		frappe.cache().hset("Supplier", "Tags", tags)
+	filters = {"document_type": "Supplier"}
+	tags = list(set(tag.tag for tag in frappe.get_all("Tag Link", filters=filters, fields=["tag"]) if tag))
 
+	if tags and tags != frappe.cache().hget("Supplier", "Tags"):
+		frappe.cache().hset("Supplier", "Tags", tags)
 	return frappe.cache().hget("Supplier", "Tags")
```

Comment:
```suggestion
	return tags
```

This cache is kinda useless now. Why not just return the output directly?
The root cause from what I can tell is lack of cache invalidation when new tags are added on Supplier doctype. 

This is a simple select query and is not used that often.  IMO it's better to get rid of caching completely here.